File: card-core/src/main/java/com/jean202/cardmizer/core/domain/RecommendationCandidate.java
package com.jean202.cardmizer.core.domain;

import java.util.Objects;

public record RecommendationCandidate(Card card, String reason, int score) {
    public RecommendationCandidate {
        Objects.requireNonNull(card, "card must not be null");
        if (reason == null || reason.isBlank()) {
            throw new IllegalArgumentException("Reason must not be blank");
        }
        if (score < 0) {
            throw new IllegalArgumentException("Score must not be negative");
        }
    }
}

File: card-core/src/main/kotlin/com/jean202/cardmizer/core/domain/RecommendationCandidate.kt
package com.jean202.cardmizer.core.domain

data class RecommendationCandidate(
    val card: Card,
    val reason: String,
    val score: Int,
) {
    init {
        require(reason.isNotBlank()) { "Reason must not be blank" }
        require(score >= 0) { "Score must not be negative" }
    }
}
